### DISCUSSION TITLE
Silk.NET/docs/hlu/troubleshooting

### DISCUSSION THREAD

**Discussion by giscus** [dotnet/Silk.NET#1620] [Webpage Comments]
# Silk.NET/docs/hlu/troubleshooting

Your one-stop-shop for high performance .NET graphics & compute. A .NET Foundation project.

https://dotnet.github.io/Silk.NET/docs/hlu/troubleshooting.html

**Comment by Perksey** [comment]
We should add guidance here for using Native AOT... (also we need to fix the tables rofl)

**Reply by joshwkearney** [reply]
Are there any tricks to getting this to work with AOT? I'm on linux and tried linking glfw in the .csproj and tried manually loading it with NativeLibrary.Load but no luck

**Reply by Vercidium** [reply]
+1, doesn't work when publishing with Native AOT on .NET 8.0.

It publishes with trim warnings:
```
.nuget\packages\silk.net.windowing.common\2.21.0\lib\netstandard2.1\Silk.NET.Windowing.Common.dll : warning IL2104: Assembly 'Silk.NE
T.Windowing.Common' produced trim warnings
```

There's a `glfw3.dll` file in the publish directory, but still has this error on startup:

```
Unhandled Exception: System.PlatformNotSupportedException: Couldn't find a suitable window platform. (none registered) https://dotnet.github.io/Silk.NET/docs/hlu/troubleshooting.html
   at Silk.NET.Windowing.Window.Create(WindowOptions) + 0xa7
```

**Reply by Perksey** [reply]
In lieu of a page update, if you add `GlfwWindowing.Use` and `GlfwInput.Use` at the start of your program, it should work.
